flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -361,7 +361,8 @@ public void onContainersAllocated(List<Container> containers) {
 					"Received new container: {} - Remaining pending container requests: {}",
 					container.getId(),
 					numPendingContainerRequests);
-
+				resourceManagerClient.removeContainerRequest(new AMRMClient.ContainerRequest(
```

Comment:
maybe we can move this removeContainerRequest before this for loop, use a indepedent for loop to do this remove staff.

```suggestion
				resourceManagerClient.removeContainerRequest(new AMRMClient.ContainerRequest(
```